src/fallback.rs

```diff
@@ -0,0 +1,346 @@
+// This module defines pure Rust platform independent implementations of all
+// the memchr routines. We do our best to make them fast. Some of them may even
+// get auto-vectorized.
+
+use core::cmp;
+use core::ptr;
+use core::usize;
+
+#[cfg(target_pointer_width = "32")]
+const USIZE_BYTES: usize = 4;
```

Comment:
Yeah, I actually made it a compile error to build memchr on 16 bit platforms because I don't have an easy way to test it: https://github.com/BurntSushi/rust-memchr/pull/35/files#diff-b4aea3e418ccdb71239b96952d9cddb6R31